services/hidadapter.md

```diff
@@ -0,0 +1,89 @@
+# HID Adapter
+
+    identifier: 0x1e5758b5
+    
+A service for configuring how Jacdac device map to HID input events. Users can have multiple configurations and swap between them by writing to `current_configuration`.
+
+## Registers
+
+    rw num_configurations: u8 @ 0x80
+    
+The number of configurations stored on the server.
+
+    rw current_configuration: u8 @ 0x81
+
+The current configuration the server is using.
+
+## Commands
+
+    command get_configuration @ 0x80 {
+        results: pipe
+        configuration_number: u8
+    }
+    pipe report configuration {
```

Comment:
@mmoskal is there a way I can reuse this report definition? we use it for both get and set!

Reply:
unfortunately no